docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,618 @@
+  input_ids = tokenizer.encode(input_string, return_tensors="pt")
+  res = hfmodel.generate(input_ids, **generator_args)
+  output = tokenizer.batch_decode(res, skip_special_tokens=True)
+  output = [item.split("<sep>") for item in output]
+  return output
+```
+
+Using it to create a question for the given text as :
+
+```python
+text_passage = "What is Qdrant? [Qdrant](http://qdrant.tech) is a vector similarity search engine that provides a \
+production-ready service with a convenient API to store, search, and manage points (i.e. vectors) with an additional payload.\
+You can think of the payloads as additional pieces of information that can help you hone in on your search and also receive \
+useful information that you can give to your users. You can get started using Qdrant with the Python `qdrant-client`, \
+by pulling the latest docker image of `qdrant` and connecting to it locally, or by trying out [Qdrant's Cloud](https://cloud.qdrant.io/) \
+free tier option until you are ready to make the full switch. With that out of the way, let's talk about what are vector databases."
+
+hf_run_model(text_passage)
+
+##Outputs 
+#[['What is Qdrant?',
+# ' How can you get started using Qrant with the Python qdrant-client']]
+```
+
+As you can see it gives you a pretty sensible set of questions, provided you can surface the answers manually.
+
+
+#### 2) **Method 2 -  Using OpenAI.**
+
+```python
+import openai
+from getpass import getpass
+import os
+
+if not (OPENAI_API_KEY := os.getenv("OPENAI_API_KEY")):
+    OPENAI_API_KEY = getpass("🔑 Enter your OpenAI API key: ")
+openai.api_key = OPENAI_API_KEY
+os.environ["OPENAI_API_KEY"] = OPENAI_API_KEY
+
+model ="gpt-3.5-turbo"
+client = OpenAI()
+
+ # Function to generate synthetic question-answer pairs
+def generate_question_answer(context):
+
+  prompt = f"""Generate a question and answer pair based keeping in mind the following: 
+  Please generate a clear and concise question that requires understanding of the content provided in the document chunk. Ensure that the question is specific, relevant, and not too broad.
+  Avoid questions such as 'in the given passage or document chunk' kind of questions. Ensure the question is about the concept the document chunk is about.
+  Provide a complete , detailed and accurate answer to the question. Make sure that the answer addresses the question directly and comprehensively, drawing from the information provided in the document chunk.
+  Use technical terminology appropriately and maintain clarity throughout the response.
+  Based on this Context : {context}
+  """
+        
+  response = client.chat.completions.create(
+        model = model,
+        messages = [ 
+        {'role': 'user', 'content': prompt}
+          ],
+        temperature=0.5,
+        max_tokens=200,
+        n=1,
+        stop=None
+    )
+  return response.choices[0].message.content
+```
+
+Using the subroutine above to generate question and answer/ground-truth pairs as :
```

Comment:
I'd advise against using the word subroutine for a function (while I do realize functions are subroutines, they are also objects and blocks of code as well, but these are less clear :) )